RFCs/FS-1004-result-type.md

```diff
@@ -48,3 +48,42 @@ The standard range of testing considerations for F# library types apply.
 - declare as struct
 - compiled name
 - require additional functions?
+
+### Proposal for additional functions
+
+Minimal:
+
+```fsharp
+bind : ('TSuccess -> Result<'T, 'TError>) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+map : ('TSuccess -> 'T) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
+```
+
+Nice-to-have:
+
+```fsharp
+fold : ('TSuccess -> 'T) -> ('TError -> 'T) -> Result<'TSuccess, 'TError> -> 'T
+mapError : ('TError -> 'T) -> Result<'TSuccess, 'TError> -> Result<'TSuccess, 'T>
+bimap : ('TSuccess -> 'T) -> ('TError -> 'U) -> Result<'TSuccess, 'TError> -> Result<'T, 'U>
+// `bimap` or `mapBoth`?
+```
+
+Low priority -- should we consider the following to get some parity with the `Option` module?
+
+```fsharp
+iter : ('TSuccess -> unit) -> Result<'TSuccess, 'TError> -> unit
+toArray : Result<'TSuccess, 'TError> -> 'TSuccess []
+toList : Result<'TSuccess, 'TError> -> 'TSuccess list
```

Comment:
I don't really like any of these ``toList``, ``toArray`` or ``toSeq``, partly because information is lost if a ``Error`` case occurred (all you get is an empty collection).  An explicit match would be clearer in these cases and force the user to do something explicit with the ``Error`` case.